### PR TITLE
Javascript helper to check for classes on all subsequent parent elements

### PR DESCRIPTION
**Pull request by sammorrisdesign-zz**
As part of the weather app we needed to detect clicks on a certain element but ended up, occasionally, getting one of its children instead. From a few quick discussions it seems like this is something we should have globally available. So I wrote a helper and tests for it. Javascript!

cc @phamann @ironsidevsquincy @rich-nguyen 

i-have-no-idea-what-im-doing-1

### CODE REVIEW

**Comment by robertberry** [comment]
Why not just use bonzo's isAncestor? https://github.com/ded/bonzo#api-isAncestor

File: static/test/javascripts/spec/common/utils/has-parent.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        });
+
+        it('should find parent class', function() {
+            expect(hasParent('container', qwery('.button__label'))).toBeTruthy();
+        });
+
+        it('should fail to find matching parent class', function() {
+            expect(hasParent('fake-class', qwery('.button__label'))).toBeFalsy();
+        });
+    });
```
Comment (ironsidevsquincy) [inline static/test/javascripts/spec/common/utils/has-parent.spec.js]: TESTS!!!!!!!!!!!! :heart: :heart: :heart: :heart: :heart: :heart:

**Comment by ironsidevsquincy** [comment]
Or https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/utils/$.js#L17

**Comment by sammorrisdesign** [comment]
tumblr_lzl4aapwb91r9ekmao1_250

File: static/src/javascripts/projects/common/utils/has-parent.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+], function (
+    qwery
+) {
+
+    function hasParent(selector, context) {
+        while (!context[0].classList.contains(selector)) {
+            if (context[0].tagName === 'HTML') {
+                return false;
+            } else {
+                context = qwery(context[0].parentElement);
```
Comment (sndrs) [inline static/src/javascripts/projects/common/utils/has-parent.js]: does this need to be `qwery`d? can you not just set `context` to the `parentElement`?

File: static/src/javascripts/projects/common/utils/has-parent.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+define([
+    'qwery'
+], function (
+    qwery
+) {
+
+    function hasParent(selector, context) {
+        while (!context[0].classList.contains(selector)) {
```
Comment (sndrs) [inline static/src/javascripts/projects/common/utils/has-parent.js]: if the first check is for the selector to apply to the context, i think you'll get a false positive for elements for which that is `true`, but for who's parents it's `false`

**Comment by sndrs** [comment]
maybe utils should be stripped out into individual modules? it's a bit hard to know there's a similar method already. that way, you could at least see a relevant looking file name...

**Comment by robertberry** [comment]
I think it's impossible to avoid occasionally writing something you didn't realise already existed. Sorry, @sammorrisdesign - code looks good, though. & tests!

**Pull request closed without merging**